File: Objective_C/Projects/Phase1_Exam/Phase1_Exam/Person.m
//
//  Person.m
//  Phase1_Exam
//
//  Created by kongyunpeng on 12/27/15.
//  Copyright © 2015 kongyunpeng. All rights reserved.
//

#import "Person.h"

@implementation Person

-(void)setName:(NSString *)name{
    _name = name;
}
-(NSString *)name {
    return _name;
}
-(void)setAge:(NSInteger)age {
    _age = age;
}
-(NSInteger)age {
    return _age;
}
-(void)setIsMarried:(BOOL)isMarried {
    _isMarried = isMarried;
}
-(BOOL)isMarried {
    return _isMarried;
}

-(void)print {
    NSLog(@"the name is :%@,age%li,isMarried%i",_name,_age,_isMarried);
}
-(instancetype)initWithName:(NSString *)name age:(NSInteger)age isMarried:(BOOL)isMarried {
    self = super.init;
    if (self) {
        _name = name;
        _age = age;
        _isMarried = isMarried;
    }
    return self;
}
@end

File: Swfit/Projects/Initialization/Initialization/Person.swift
//
//  Person.swift
//  Initialization
//
//  Created by kongyunpeng on 12/26/15.
//  Copyright © 2015 kongyunpeng. All rights reserved.
//


class Person {
    var name:String
    var age:Int
    var sex:Bool
    init(name:String,age:Int,sex:Bool) {
        self.name = name
        self.age = age
        self.sex = sex
    }
    /*
    Safety check 3
    A convenience initializer must delegate to another initializer before assigning a value to any property (including properties defined by the same class). If it doesn’t, the new value the convenience initializer assigns will be overwritten by its own class’s designated initializer.*/
    convenience init() {
        self.init(name:"Jay",age:1,sex:true)
        self.name = "Jay"
        
    }
    
    
}
